engine/src/main/java/org/terasology/world/block/entity/BlockEntitySystem.java

```diff
@@ -88,12 +87,12 @@ public void doDestroy(DoDestroyEvent event, EntityRef entity, ActAsBlockComponen
     @ReceiveEvent(priority = EventPriority.PRIORITY_LOW)
     public void doDestroy(DoDestroyEvent event, EntityRef entity, BlockComponent blockComponent) {
         commonDestroyed(event, entity, blockComponent.block);
-        worldProvider.setBlock(blockComponent.position, blockManager.getBlock(BlockManager.AIR_ID));
+        worldProvider.setBlock(blockComponent.getPosition(new Vector3i()), blockManager.getBlock(BlockManager.AIR_ID));
```

Comment:
This makes we wonder whether `BlockComponent` should just expose `Vector3ic getPosition()` for read-only access 🤔

Reply:
that's ideally what I would like to do.

Reply:
I wouldn't do it in this PR, but I might just make another PR for that today (after getting through some more JOML PRs 🙃 )

LGTM as is 👍